loadtest/loadtest.go

```diff
@@ -99,17 +114,30 @@ func (lt *LoadTester) Run() error {
 
 // Stop terminates the current load-test
 func (lt *LoadTester) Stop() error {
-	if !lt.started {
+	if !lt.hasStarted() {
 		return ErrNotRunning
 	}
-	for range lt.controllers {
+
+	var controllers []control.UserController
+	lt.controllersMut.RLock()
+	controllers = lt.controllers
+	lt.controllersMut.RUnlock()
+	for range controllers {
 		if err := lt.RemoveUser(); err != nil {
 			mlog.Error(err.Error())
 		}
 	}
 	lt.wg.Wait()
-	close(lt.status)
+	lt.startedMut.Lock()
 	lt.started = false
+	lt.startedMut.Unlock()
```

Comment:
This way we wouldn't guard against two goroutines attempting to remove users at the same time. While it wouldn't cause a race condition cause of the internal locks in `RemoveUser` I don't think it should be allowed. 
Maybe we should add a `stopping` variable/channel to make sure everything in `Stop` is run just by one goroutine and also avoid any other API call during this teardown phase. Thoughts?

Reply:
Do you mean since we are reading a copy of `controllers` and then removing users for the length of that slice, there may be a possibility that 2 goroutines get the same length and one of them return a `ErrNoUsersLeft` ?

You want to combine the controllersMut and startedMut in a single larger mutex ? But wouldn't that still cause racy access for lt.controllers now that it won't be controlled by the same mutex ?

Reply:
> Do you mean since we are reading a copy of `controllers` and then removing users for the length of that slice, there may be a possibility that 2 goroutines get the same length and one of them return a `ErrNoUsersLeft` ?

Yes, only one goroutine should be able to stop a running load-test. This is more about keeping integrity and make the API behave as expected.

> You want to combine the controllersMut and startedMut in a single larger mutex ? But wouldn't that still cause racy access for lt.controllers now that it won't be controlled by the same mutex ?

Not exactly. What I think we need here is an additional state variable (atomic) or channel (whatever comes the simplest to implement).
We call it `stopping` and use it to make sure that only one goroutine is allowed to run `Stop()` and also while that call is running no other goroutine is allowed to make any API call that touches the internal state (Start/Stop/AddUser/RemoveUser will have to check for both `started ` and this new variable to go through).
This would also remove the need for the `statusClosedMut` since the `close` would be guaranteed to happen only once.